main.c

```diff
@@ -30,6 +30,37 @@
 #include "wlr-screencopy-unstable-v1-client-protocol.h"
 #include "ext-session-lock-v1-client-protocol.h"
 
+#define MAX_OUTPUT_SIZE 1024
+
+static struct swaylock_state state;
+
+void update_script() {
+	FILE *pipe;
+	char path[1035];
+	char output[MAX_OUTPUT_SIZE];
+	char *script_str = (char *)malloc((MAX_OUTPUT_SIZE + 1) * sizeof(char));
+
+	pipe = popen(state.args.script_path, "r");
+
+	if (pipe == NULL) {
+		fprintf(stderr, "Failed to run script.\n");
```

Comment:
this project has a `swaylock_log_errno` defined somewhere, and `popen` sets `errno` - does the file not exist? are its permissions insufficient? etc